pkg/pdutil/api_client.go

```diff
@@ -128,6 +130,9 @@ func (pc *pdAPIClient) ListGcServiceSafePoint(
 		err  error
 	)
 	err = retry.Do(ctx, func() error {
+		ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
It's better to define a constant for this magic number because we hardcode it twice.